lace/emulator/pd_archive.py

```diff
@@ -0,0 +1,920 @@
+        # we store the power of each phase and axis separately
+        self.data = []
+
+        # keys in output dictionary
+        keys_copy_in = [
+            "mF",
+            "sim_T0",
+            "sim_gamma",
+            "sim_sigT_Mpc",
+            "kF_Mpc",
+            "k_Mpc",
+            "p1d_Mpc",
+            "scale_tau",
+            "sim_scale_T0",
+            "sim_scale_gamma",
+        ]
+        keys_copy_out = [
+            "mF",
+            "T0",
+            "gamma",
+            "sigT_Mpc",
+            "kF_Mpc",
+            "k_Mpc",
+            "p1d_Mpc",
+            "scale_tau",
+            "scale_T0",
+            "scale_gamma",
+        ]
+        # we also have P3D data for the 768 post-processing
+        if self.post_processing == "768":
+            keys_copy_in.append("p3d_data")
+            keys_copy_out.append("k3_Mpc")
+            keys_copy_out.append("mu3")
+            keys_copy_out.append("p3d_Mpc")
+
+        # read file containing information about latin hyper-cube
+        cube_json = self.fulldir + "/latin_hypercube.json"
+        with open(cube_json) as json_file:
+            self.cube_data = json.load(json_file)
+        if self.verbose:
+            print("latin hyper-cube data", self.cube_data)
+        if nsamples is None:
+            self.nsamples = self.cube_data["nsamples"]
+        else:
+            self.nsamples = nsamples
+        if self.verbose:
+            print("simulation suite has %d samples" % self.nsamples)
+
+        # read pivot point from simulation suite if not specified
+        if self.kp_Mpc is None:
+            n_star = self.cube_data["param_space"]["n_star"]
+            self.kp_Mpc = n_star["kp_Mpc"]
+            update_kp = False
+        elif self.kp_Mpc == self.cube_data["param_space"]["n_star"]["kp_Mpc"]:
+            ## If selected k_p is same as in the archive, do not recompute
+            update_kp = False
+        else:
+            # will trigger slow code, could check that kp has indeed changed
+            update_kp = True
+
+        if pick_sim is not None:
+            if np.issubdtype(type(pick_sim), np.integer):
+                start = pick_sim
+                self.nsamples = pick_sim + 1
+            else:
+                start = 0
+                self.nsamples = 1
+        else:
+            start = 0
+
+        # read info from all sims, all snapshots, all rescalings
+        for sample in range(start, self.nsamples, undersample_cube):
+            if sample is drop_sim:
+                continue
+            elif (sample == 30) & (drop_sim is "central"):
+                continue
+            # store parameters for simulation pair / model
+            sim_params = self.cube_data["samples"]["%d" % sample]
+            if self.verbose:
+                print(sample, "sample has sim params =", sim_params)
+
+            # read number of snapshots (should be the same in all sims)
+            ind_sim = sample
```

Comment:
Wait, `ind_sim` is the number of snapshots, or is it an ID for each simulation pair?